fix: match Juju's action name validation rules (#2723)
canonical/charmcraft: docs/release-notes/charmcraft-4.2.rst

.. _release-4.2:

> .. _meta::
>     :description: Release notes for Charmcraft 4.2, listing the new features, behavioural changes, deprecations, and bug fixes shipped in the 4.2.x patch series.

Reviewer: ```suggestion
.. meta::
    :description: Learn about the new features, changes, and fixes introduced in Charmcraft 4.2.
```
We've just been reusing the page's first sentence for release notes meta descriptions.

Author: This one I will leave alone, because I assume there will be / is already one of these and so it should get merged out of the PR. But I had to include it to tick all the boxes.

- `#2598 <https://github.com/canonical/charmcraft/issues/2598>`__ Remote build fails
when getting project name

> - `#TODO <https://github.com/canonical/charmcraft/pull/TODO>`__ Match Juju's

Author: ```suggestion
- `#2723 <https://github.com/canonical/charmcraft/pull/2723>`__ Match Juju's
```
